tests/db/test_crud.py

```diff
@@ -0,0 +1,44 @@
+""" Mocking calls to database for functional testing """
+from pytest_bdd import scenario, given, then
+from sqlalchemy import DateTime, cast
+# TODO: Is this even a good one to use?
+from alchemy_mock.mocking import UnifiedAlchemyMagicMock
+from db import crud, models
+import schemas
+
+
+@scenario('test_crud.feature', 'Create a single forecast',
+          example_converters=dict(
+              issue_date=str,
+              date_time=str,
+              temperature=float,
+              rh=float,
+              wind_speed=float,
+              total_precip=float,
+              station_code=int
+          ))
+def test_create_single_forecast():
+    # NOTE: Once we do CRUD through the API, relook this test.
+    """ BDD Scenario. """
+
+@given("I am a session creating a forecast with <date_time>, <temperature>, <rh>, <wind_speed>, <total_precip> at time <issue_date> for station <station_code>")
+def create_single_forecast(date_time: str, temperature: float, rh: float, wind_speed: float,
+                           total_precip: float, issue_date: str, station_code: int):
+    """ Create a single forecast & mock adding it to the DB """
+    session = UnifiedAlchemyMagicMock()  # TODO: How far do we take this?
```

Comment:
Does this UnifiedAlchemyMagicMock require any clean up after each transaction?

Reply:
nope! Because it's only mocking transactions with the database, not actually making them, there's no cleanup required :)